optaplanner-examples/src/main/java/org/optaplanner/examples/curriculumcourse/domain/solver/LectureDifficultyWeightFactory.java

```diff
@@ -22,11 +22,13 @@
 import org.optaplanner.examples.curriculumcourse.domain.CourseSchedule;
 import org.optaplanner.examples.curriculumcourse.domain.Lecture;
 import org.optaplanner.examples.curriculumcourse.domain.UnavailablePeriodPenalty;
+import org.optaplanner.examples.curriculumcourse.domain.solver.LectureDifficultyWeightFactory.LectureDifficultyWeight;
 
-public class LectureDifficultyWeightFactory implements SelectionSorterWeightFactory<CourseSchedule, Lecture> {
+public class LectureDifficultyWeightFactory
+        implements SelectionSorterWeightFactory<CourseSchedule, Lecture, LectureDifficultyWeight> {
 
     @Override
-    public Comparable createSorterWeight(CourseSchedule schedule, Lecture lecture) {
+    public LectureDifficultyWeight createSorterWeight(CourseSchedule schedule, Lecture lecture) {
```

Comment:
Returning LectureDifficultyWeight is a good thing here. That can be done with or without adding that 3 generic parameter. So even if we withold this PR, we should still do that.